src/Tools/ExternalAccess/Debugger/GlassTestsHotReloadService.cs

```diff
@@ -0,0 +1,121 @@
+            _debuggerService = debuggerService;
+        }
+
+        public async Task StartSessionAsync(Solution solution, CancellationToken cancellationToken)
+        {
+            var newSessionId = await _encService.StartDebuggingSessionAsync(
+                solution,
+                new ManagedHotReloadServiceImpl(_debuggerService),
+                captureMatchingDocuments: ImmutableArray<DocumentId>.Empty,
+                captureAllMatchingDocuments: true,
+                reportDiagnostics: false,
+                cancellationToken).ConfigureAwait(false);
+
+            Contract.ThrowIfFalse(_sessionId == default, "Session already started");
+            _sessionId = newSessionId;
+        }
+
+        public void EnterBreakState()
+        {
+            var sessionId = _sessionId;
+            Contract.ThrowIfFalse(sessionId != default, "Session has not started");
+
+            _encService.BreakStateOrCapabilitiesChanged(sessionId, inBreakState: true, out _);
+        }
+
+        public void ExitBreakState()
+        {
+            var sessionId = _sessionId;
+            Contract.ThrowIfFalse(sessionId != default, "Session has not started");
+
+            _encService.BreakStateOrCapabilitiesChanged(sessionId, inBreakState: false, out _);
+        }
+
+        public void OnCapabilitiesChanged()
+        {
+            var sessionId = _sessionId;
+            Contract.ThrowIfFalse(sessionId != default, "Session has not started");
+
+            _encService.BreakStateOrCapabilitiesChanged(sessionId, inBreakState: null, out _);
+        }
+
+        public void CommitSolutionUpdate()
+        {
+            var sessionId = _sessionId;
+            Contract.ThrowIfFalse(sessionId != default, "Session has not started");
+
+            _encService.CommitSolutionUpdate(sessionId, out _);
+        }
+
+        public void DiscardSolutionUpdate()
+        {
+            var sessionId = _sessionId;
+            Contract.ThrowIfFalse(sessionId != default, "Session has not started");
+
+            _encService.DiscardSolutionUpdate(sessionId);
+        }
+
+        public void EndDebuggingSession()
+        {
+            var sessionId = _sessionId;
+            Contract.ThrowIfFalse(sessionId != default, "Session has not started");
+
+            _encService.EndDebuggingSession(sessionId, out _);
+            _sessionId = default;
+        }
+
+        public async ValueTask<bool> HasChangesAsync(Solution solution, string? sourceFilePath, CancellationToken cancellationToken)
+        {
+            var sessionId = _sessionId;
+            if (sessionId == default)
+            {
+                return false;
+            }
+
+            return await _encService.HasChangesAsync(sessionId, solution, s_noActiveStatementSpanProvider, sourceFilePath, cancellationToken).ConfigureAwait(false);
+        }
+
+        public async ValueTask<ManagedHotReloadUpdates> EmitSolutionUpdateAsync(Solution solution, CancellationToken cancellationToken)
```

Comment:
>EmitSolutionUpdateAsync [](http://example.com/codeflow?start=56&length=23)

EditAndContinueLanguageService has two methods 
GetHotReloadUpdatesAsync and GetEditAndContinueUpdatesAsync that are slightly different (the return type and the way how active statements are handled). I think we need both.

Reply:
Added. I wasn't sure about this, because without the active statements its just the return type, but it doesn't hurt to add.

Reply:
You're right, the active statement spans are the main difference, but the debugger has two code paths currently, so they would likely want to get the results in the right shape.